### ISSUE TITLE
Documentation example doesn't work with TypeScript

### ISSUE THREAD

**Issue by filipw01** [kettanaito/msw#1454]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Reproduction repository

https://stackblitz.com/edit/node-ifch93?file=index.ts

### Reproduction steps

Use documentation example with typescript https://mswjs.io/docs/getting-started/integrate/browser#configure-worker

```
// src/mocks/browser.js
import { setupWorker } from 'msw'
import { handlers } from './handlers'

// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...handlers)
```

### Current behavior

Typescript throws multiple 
`TS4094: Property 'SOMETHING' of exported class expression may not be private or protected.`

### Expected behavior

I expect typescript to be fine with that

**Comment by gduliscouet-ubitransport** [comment]
I have the same issue (with `typescript 4.7.4`).

Here is an explaination of the warning from https://github.com/microsoft/TypeScript/issues/30355:

> "Exported anonymous classes can't have private or protected members if declaration emit is enabled, because there's no way to represent that in a .d.ts file"

I'm trying this workaround in the meanwhile, taken from [here](https://github.com/microsoft/TypeScript/issues/30355#issuecomment-1320168471) (not sure I'm doing it well, I didn't check everything)

```ts
import { setupWorker, type SetupWorkerApi } from 'msw'
import { handlers } from './handlers'

type SetupWorkerApiWithoutPrivate = Omit<SetupWorkerApi, never>

// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...handlers) as SetupWorkerApiWithoutPrivate
```

**Comment by kettanaito** [comment]
I'm not sure how this is related to MSW as we don't emit anonymous classes. You can see that the `SetupServerApi` class is named: 

- [built](https://unpkg.com/browse/msw@0.49.0/lib/SetupServerApi-3406bf53.d.ts)
- source:
https://github.com/mswjs/msw/blob/2553156fcf47a0c879f5f2d1f43bebeca361d684/src/node/SetupServerApi.ts#L31

This must be something else. The library builds without issues for me on TS 4.8 but I haven't used it in an example project to verify the consumer's side.

**Comment by gduliscouet-ubitransport** [comment]
You are right, the issue I shared is talking about an "anonymous class". But here you can see the same TS error with a named class: https://github.com/Microsoft/TypeScript/issues/17293.

From what I understand, you should be able to do that, but you cannot due to some limitations in the declaration file creation. See this issue regrouping all the issues with .d.ts declaration files: https://github.com/microsoft/TypeScript/issues/35822.

From what I see, the only solution seems to not use `private` yet in a typescript lib, or apply some workaround on the exported types :/

You can launch the repro of @filipw01 to see the error on consumer side: https://stackblitz.com/edit/node-ifch93?file=index.ts

**Comment by andykenward** [comment]
@gduliscouet-ubitransport Adding the type annotation to the `worker` constant instead of using TypeScript inference allows me to output the definition file without errors.
```ts
import { setupWorker, type SetupWorkerApi } from 'msw';

export const worker:SetupWorkerApi = setupWorker(...[]);
```

**Comment by gduliscouet-ubitransport** [comment]
I confrim ! Thank you @andykenward for sharing.

In fact I realised that:
- the `SetupWorkerApi` that is used to define the return of `setupWorker` comes from `index.d.ts` and is a `class` with `private` members
- the `SetupWorkerApi` that we import comes from `SetupApi-0d3126ba.d.ts`  and it is an `interface`. That's why it doesn't break (the `Omit` I did above was useless)

![image](https://user-images.githubusercontent.com/101416921/203943270-cf8a295f-8ef7-4f61-b6cd-c9d5666a8d24.png)

![image](https://user-images.githubusercontent.com/101416921/203943362-965862bc-2b89-436f-875f-0e974746d683.png)

**Comment by gduliscouet-ubitransport** [comment]
@kettanaito I opened a PR to reuse the existing interface describing the class `SetupWorkerApi`. For me it is the only way to have it working on the consumer side (without the above workaround), but maybe you want to check the interface `SetupWorkerApi` is describing well enough the class `SetupWorkerApi` (at least TS it is matching according to TS)

**Comment by kettanaito** [comment]
Thanks for adding the fix, @gduliscouet-ubitransport! 

I left a few comments on how we could export both the type and the class and fix this issue. Hopefully, that direction would also work.

**Comment by kettanaito** [comment]
## Released: v0.49.1 🎉

This has been released in v0.49.1!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.49.1)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.49.1)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
